design/baremetal-operator/bulk-set-bios-config.md

```diff
@@ -0,0 +1,215 @@
+a schema for the BIOS settings), and a bulk BIOS settings for Ironic to write
+to the BMC. The current BIOS settings and BIOS Registry can be used to create
+the bulk BIOS settings which can be set on multiple hosts of the same type,
+the current settings and registry will not be used on these hosts.
+
+The current BIOS settings are retrieved from the BMC by Ironic and cached
+whenever the node moves to `manageable` or `cleaning`. These BIOS settings
+can be read via the Ironic API (`/v1/nodes/{node_ident}/bios`) as a set of
+name:value pairs. Ironic can set the BIOS Configuration values through the use
+of clean_steps as part of the manual cleaning process as described in the Ironic
+documentation. BIOS settings are configured as name/value pairs and can include
+from one to multiple pairs.
+
+What we are proposing is to make the cached BIOS settings from Ironic available
+to the user via a ConfigMap referenced in the BareMetalHostStatus.  As there
+are no fields in BIOS configuration that are secret (i.e. no passwords etc.)
+this new entry does not need to be secret-protected. 
+
+A new Ironic API must be provided for the attributes section of the BIOS
+registry (`/v1/nodes/{node_ident}/bios_registry`) as this is currently not
+supported in Ironic. The registry contains a JSON schema describing the BIOS
+settings - names, types, enumeration values, readonly flag etc. The BIOS
+registry will be retrieved from the BMC by Ironic at the same time Ironic
+gets the BIOS settings, as described in the Ironic proposal
+[BIOS Registry](https://review.opendev.org/c/openstack/ironic-specs/+/774681).
+An example of an attributes entry in the BIOS registry is:
+
+    {
+        "AttributeName": "SriovGlobalEnable",
+        "CurrentValue": null,
+        "DisplayName": "SR-IOV Global Enable",
+        "DisplayOrder": 2331,
+        "HelpText": "Enables or disables the BIOS configuration of Single Root
+                     I/O Virtualization (SR-IOV) devices. Enable this feature
+                     if booting to a virtualization operating system that
+                     recognize SR-IOV devices. ",
+        "Hidden": false,
+        "Immutable": false,
+        "MenuPath": "./IntegratedDevicesRef",
+        "ReadOnly": false,
+        "ResetRequired": true,
+        "Type": "Enumeration",
+        "Value": [
+          {
+            "ValueDisplayName": "Enabled",
+            "ValueName": "Enabled"
+          },
+          {
+            "ValueDisplayName": "Disabled",
+            "ValueName": "Disabled"
+          }
+        ],
+        "WarningText": null,
+        "WriteOnly": false
+    }
+
+The cached BIOS registry from Ironic will also be made available in the
+BaremetalHost CRD. This can be a ConfigMap referenced by the
+`BaremetalHostStatus`. Since the BIOSRegistry can be quite large this should be
+stored gzip compressed so as to not exceed the 1Mb data limit in the ConfigMap.
```

Comment:
These are big, and likely to be the same for a lot of hosts in a homogeneous environment. Instead of always collecting them, what would an API look like for requesting a registry explicitly?

Reply:
Good point, we only need the BIOS registry on the reference hosts, there is no reason to store it on all hosts in particular, especially considering the size of this registry.  I've updated this section although left the API definition general for the design phase.

Reply:
Got it :-)  I'll work on adding details here.